Updated the date filter

ningxie1991 opened the pull request:
For #230

This is part of the changes to improve the date range filter. 

Changes:
1. Moved date filter from top bar to the header control of the focus page
2. Focus page header controls will not be refreshed on re-render
3. Date picker implementation includes the original custom date ranges and user chosen dates
4. Updated explore url to include user-chosen date ranges

Demo video for version 1:
*Tip: For selecting one day, just double click that day to create a range from=chosenDay&to=chosenDay

https://user-images.githubusercontent.com/3387698/135055140-113e36b1-262d-4694-9e51-bf86e9d08de3.mp4


Demo video for version 2:

https://user-images.githubusercontent.com/3387698/135055027-5aaa3d25-7bec-4324-b864-c9ff50debbdd.mp4



New look:

![image](https://user-images.githubusercontent.com/3387698/135452817-ed453a01-e515-441b-9596-b066957790a8.png)

ningxie1991 committed:
Updated the date filter

chaoran-chen commented:
Hi Ning. I just checked out the preview and it works great! Just one suggestion: I am a bit afraid that some people are not going to find the date pickers or not notice that they are not seeing the full data (e.g., when they just click on a link). What do you think about putting the date picker to the top of the right area:

![image](https://user-images.githubusercontent.com/18666552/134656635-4f956abd-fb39-4102-a7e0-476296bdc84e.png)

chaoran-chen commented:
We need to settle our package.json-issue! ;) You and @TKGZ are alternating with adding and deleting (see https://github.com/cevo-public/cov-spectrum-website/commit/73aa25830dbdff15bc88b7ddfb678449eae67a2f) 26k rows.

ningxie1991 commented:
> Hi Ning. I just checked out the preview and it works great! Just one suggestion: I am a bit afraid that some people are not going to find the date pickers or not notice that they are not seeing the full data (e.g., when they just click on a link). What do you think about putting the date picker to the top of the right area:
> 
> ![image](https://user-images.githubusercontent.com/18666552/134656635-4f956abd-fb39-4102-a7e0-476296bdc84e.png)


Yeah I agree that we could make it more obvious. How about this?

https://user-images.githubusercontent.com/3387698/134689049-985916f9-91b5-4a0e-a7b6-2750c8713c5d.mp4



![image](https://user-images.githubusercontent.com/3387698/134690168-174ae5c6-1df3-4626-8672-43a4763d6ce0.png)

ningxie1991 commented:
> We need to settle our package.json-issue! ;) You and @TKGZ are alternating with adding and deleting (see [73aa258](https://github.com/cevo-public/cov-spectrum-website/commit/73aa25830dbdff15bc88b7ddfb678449eae67a2f)) 26k rows.

Yeah agree. I think it's the version of node or npm that's causing the difference.

ningxie1991 committed:
Updated the date filter position and look

ningxie1991 commented:
@chaoran-chen @TKGZ 
Q1: I want to disable dates before COVID, which date would be a good starting date of our data? 
Q2: I notice we use the date format yyyy-mm-dd in our API calls, but that's not the European format for dates. I am using dd/mm/yyyy in the date picker, but still keep the international format yyyy-mm-dd in the url for the api call to work. Should we make them consistent? Or rather, the question is which format should we stick to, international vs Europe?

date string format for API calls (yyyy-mm-dd):

https://github.com/cevo-public/cov-spectrum-website/blob/6177a341e456427e0a699663863f1d9df5561828/src/services/api-types.ts#L27-L28

Date filter display (dd/mm/yyyy):

![image](https://user-images.githubusercontent.com/3387698/134703809-32a769a5-e6bd-4df1-82a6-d9ef134801c1.png)

spacephoton commented:
Maybe January 1 2020? Since it’s round

I think for user best to use European? But that’s up to chairman

On Fri, 24 Sep 2021 at 17:42, ningxie ***@***.***> wrote:

> @chaoran-chen <https://github.com/chaoran-chen> @TKGZ
> <https://github.com/TKGZ>
> Q1: I want to disable dates before COVID, which date would be a good
> starting date of our data?
> Q2: I notice we use the date format yyyy-mm-dd in our API calls, but
> that's not the European format for dates. I am using dd-mm-yyyy in the date
> picker, but still keep the US format yyyy-mm-dd in the url for the api call
> to work. Should we make them consistent?
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/cevo-public/cov-spectrum-website/pull/235#issuecomment-926728339>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AIUW4NL6WXY6GCXGYPK7VIDUDSL7DANCNFSM5EVSFE2Q>
> .
>
-- 
Michael

chaoran-chen commented:
> Q1: I want to disable dates before COVID, which date would be a good starting date of our data?

We have a few sequences from end of 2019 in China but I would rather ignore them and start with the first week of 2020.

> Q2: I notice we use the date format yyyy-mm-dd in our API calls, but that's not the European format for dates. I am using dd/mm/yyyy in the date picker, but still keep the international format yyyy-mm-dd in the url for the api call to work. Should we make them consistent? Or rather, the question is which format should we stick to, international vs Europe?

In the APIs, we use the iso-format which is `yyyy-mm-dd` (https://en.wikipedia.org/wiki/ISO_8601). For the UI, what do you think about, for example, "24 Sep 2021"?

Ok, then I think it'd be better if we keep the iso-format (since it's in the url) for UI as well or spell it out like "24 Sep 2021" if we use Europe format. Let me try out both to see which is better.

ningxie1991 commented:
ISO-format:

![image](https://user-images.githubusercontent.com/3387698/134708086-8ceae1ca-acca-4440-afcb-2658ea194837.png)


Spell out European format:

![image](https://user-images.githubusercontent.com/3387698/134708223-bbc4a94f-4c0f-483d-b397-e5e2565fc676.png)


I also need to fix the scenario when user reselects some invalid range and clicks out, the input becomes blank.

chaoran-chen commented:
I like the spelled-out European variant more

chaoran-chen commented:
I think that the date in the URL does not follow the ISO 8601 format. Shouldn't it be `2021-06-14` rather than `2021-6-14`?

ningxie1991 commented:
> I think that the date in the URL does not follow the ISO 8601 format. Shouldn't it be `2021-06-14` rather than `2021-6-14`?

Yeah good point

chaoran-chen reviewed (changes requested):
Oh, I remembered another issue:

- [x] Could you please add the date filter to the sequencing intensity deep explore page?

https://cov-spectrum-develop-git-fine-grain-date-filter-cov-spectrum.vercel.app/explore/Switzerland/AllSamples/Past3M/sequencing-coverage

- [x] And since we decided to use the European format, could we please have Monday as the first day of the week?

![image](https://user-images.githubusercontent.com/18666552/134783183-309ea2d1-d27e-43a4-85da-ce2fd9b6f7fb.png)

ningxie1991 committed:
Merge branch 'develop' into fine-grain-date-filter

spacephoton commented:
Seems great overall. 
Just a few UI issues:
- [x] The alignment seems off when the screen is narrow
- [x] The design doesn't seem to fit with the other select. Might be better to make it less rounded and make it similar to the other select boxes (such as all samples above). I think mostly just rounding of edges, spacing and up/and down arrow icon. Doesn't have to be 100% same but more or less would be nice.
- [x] Color on select would be nice to have it same as other selects, such as when you focus on the main country selector

![CleanShot 2021-09-27 at 16 22 38](https://user-images.githubusercontent.com/36269621/134927501-49e0be13-0eed-430f-b46b-a24486238e67.png)

![CleanShot 2021-09-27 at 16 24 12](https://user-images.githubusercontent.com/36269621/134927790-cae9bf06-8c8a-4c10-af51-82fc4aed2b8e.png)
![CleanShot 2021-09-27 at 16 24 41](https://user-images.githubusercontent.com/36269621/134927843-7e06f960-5fee-4f6d-9858-c72dda31dd20.png)

ningxie1991 committed:
Updated the date filter

1. Start from Monday to Sunday
2. Use "27 Sep 2021" date format
3. Add date filter to sequence coverage page
4. Formatted dates from date picker with ISO format YYYY-MM-DD

ningxie1991 commented:
> Seems great overall.
> Just a few UI issues:
> 
> * [x]  The alignment seems off when the screen is narrow
> * [x]  The design doesn't seem to fit with the other select. Might be better to make it less rounded and make it similar to the other select boxes (such as all samples above). I think mostly just rounding of edges, spacing and up/and down arrow icon. Doesn't have to be 100% same but more or less would be nice.
> * [x]  Color on select would be nice to have it same as other selects, such as when you focus on the main country selector
> 
> ![CleanShot 2021-09-27 at 16 22 38](https://user-images.githubusercontent.com/36269621/134927501-49e0be13-0eed-430f-b46b-a24486238e67.png)
> 
> ![CleanShot 2021-09-27 at 16 24 12](https://user-images.githubusercontent.com/36269621/134927790-cae9bf06-8c8a-4c10-af51-82fc4aed2b8e.png)
> ![CleanShot 2021-09-27 at 16 24 41](https://user-images.githubusercontent.com/36269621/134927843-7e06f960-5fee-4f6d-9858-c72dda31dd20.png)


@TKGZ  What do you think about this positioning when screen is small? Or should I replace the date range with the filter funnel icon?

![image](https://user-images.githubusercontent.com/3387698/134936186-799a3a33-6230-4ee4-811d-87f97e9c63e1.png)

ningxie1991 committed:
Updated date filter style

corneliusroemer commented:
Hey guys! Great you're workign on this. May I give some feedback as your maybe number 1 user ;)

The functionality is great. But regarding UX I have a few opinions that you may consider or ignore:
- CovSpectrum should be usable for power users (like me): I hate date pickers. There needs to be a way to type out the date range. Right now this is not possible. Very annoying. Date pickers should be an add-on, not a first citizen.
- I'd very much prefer ISO strings in the exact format `2021-03-03` it's easier to parse mentally than mixes of strings and digits
- Proposed field content filled either by date picker or typed by user: `from:2021-03-05 until:2021-09-22`, this is the defacto world standard for human and machine readable date ranges. Everything else is unintuitive and has problems for either human or machine. Twitter API uses the above for the query type.

General feedback: I'd focus less on front end and think about usability for power users. The power users are who are deriving the biggest benefit. Everyone else just looks at colorful plots anyways. Only power users will want to change the date to specific ranges that are not pre-provided in form of `Past 3 months`

spacephoton commented:
> * CovSpectrum should be usable for power users (like me): I hate date pickers. There needs to be a way to type out the date range. Right now this is not possible. Very annoying. Date pickers should be an add-on, not a first citizen.

First of all, thanks Ning for the UI fixes, I think it looks a lot better. Not so sure about the positioning of the elements on narrow screens, will have a look later.

I think @corneliusroemer has some great points we should think about. Does indeed seem like UI main issue now, the functionality works amazing, thank you @ningxie1991 :) 

As a non power user, I don't think I ever had the urge to select specific dates, I just want to know trends.

I have to say I also don't like date pickers that much, and they mostly are good for non power users. Although I do think some power users might like them.

May I suggest some solution ideas:
Now, the most common use takes extra clicks to do, whereas the less common specific date select is easy. I think good UI makes frequently done tasks easier to do (less clicks and bigger buttons that are easier to notice). 

1. We move the multi select up here to make it easy to select, since this is the most frequently used by both power users and normal users. Specific dates seem like more of a less common case.
![CleanShot 2021-09-28 at 03 48 09@2x](https://user-images.githubusercontent.com/36269621/135009056-8272bbaf-483e-428b-99b3-8e8fb72695e9.png)

2. Turn this into simple date string? for example:  "2021-09-28 to 2021-09-29"

3. Make the time frame box editable
For the custom date range, have everything depend on the string in the text box, and optionally remove the date selector (sorry if this took a long time to implement in this case, this is just an idea, but I think we can keep). When a custom date is shown, then clicking the box enables us to edit it and also pops down the other options.
When we are on time range like "last 3 months", clicking just shows the other options, including 'custom date range' which would then let us directly edit the string.

4. Have two boxes without date selecting UI
[Starting date] -> [Ending date]
Selecting the since would give us several options: Three months ago, Week ago,  etcetera.
The
Both would have [Custom date] option.

An example would be:
[Three months ago] -> [Today]

Clicking on three months ago we could choose the 'Date string' from the dropdown, which would turn the box into an editable input. When you click the input, you can either edit the date string directly, or select another fixed option from the dropdown. 

Thoughts @chaoran-chen @ningxie1991 @corneliusroemer ?

ningxie1991 committed:
Update date range selector based on review comments

ningxie1991 commented:
> Hey guys! Great you're workign on this. May I give some feedback as your maybe number 1 user ;)
> 
> The functionality is great. But regarding UX I have a few opinions that you may consider or ignore:
> 
> * CovSpectrum should be usable for power users (like me): I hate date pickers. There needs to be a way to type out the date range. Right now this is not possible. Very annoying. Date pickers should be an add-on, not a first citizen.
> * I'd very much prefer ISO strings in the exact format `2021-03-03` it's easier to parse mentally than mixes of strings and digits
> * Proposed field content filled either by date picker or typed by user: `from:2021-03-05 until:2021-09-22`, this is the defacto world standard for human and machine readable date ranges. Everything else is unintuitive and has problems for either human or machine. Twitter API uses the above for the query type.
> 
> General feedback: I'd focus less on front end and think about usability for power users. The power users are who are deriving the biggest benefit. Everyone else just looks at colorful plots anyways. Only power users will want to change the date to specific ranges that are not pre-provided in form of `Past 3 months`

@corneliusroemer Thanks for all the feedback. Can you try out the new preview?

Demo video:

https://user-images.githubusercontent.com/3387698/135054110-8daa1b27-a1e0-47f2-b02f-2a041966d3f5.mp4

ningxie1991 commented:
> > * CovSpectrum should be usable for power users (like me): I hate date pickers. There needs to be a way to type out the date range. Right now this is not possible. Very annoying. Date pickers should be an add-on, not a first citizen.
> 
> First of all, thanks Ning for the UI fixes, I think it looks a lot better. Not so sure about the positioning of the elements on narrow screens, will have a look later.
> 
> I think @corneliusroemer has some great points we should think about. Does indeed seem like UI main issue now, the functionality works amazing, thank you @ningxie1991 :)
> 
> As a non power user, I don't think I ever had the urge to select specific dates, I just want to know trends.
> 
> I have to say I also don't like date pickers that much, and they mostly are good for non power users. Although I do think some power users might like them.
> 
> May I suggest some solution ideas:
> Now, the most common use takes extra clicks to do, whereas the less common specific date select is easy. I think good UI makes frequently done tasks easier to do (less clicks and bigger buttons that are easier to notice).
> 
> 1. We move the multi select up here to make it easy to select, since this is the most frequently used by both power users and normal users. Specific dates seem like more of a less common case.
>    ![CleanShot 2021-09-28 at 03 48 09@2x](https://user-images.githubusercontent.com/36269621/135009056-8272bbaf-483e-428b-99b3-8e8fb72695e9.png)
> 2. Turn this into simple date string? for example:  "2021-09-28 to 2021-09-29"
> 3. Make the time frame box editable
>    For the custom date range, have everything depend on the string in the text box, and optionally remove the date selector (sorry if this took a long time to implement in this case, this is just an idea, but I think we can keep). When a custom date is shown, then clicking the box enables us to edit it and also pops down the other options.
>    When we are on time range like "last 3 months", clicking just shows the other options, including 'custom date range' which would then let us directly edit the string.
> 4. Have two boxes without date selecting UI
>    [Starting date] -> [Ending date]
>    Selecting the since would give us several options: Three months ago, Week ago,  etcetera.
>    The
>    Both would have [Custom date] option.
> 
> An example would be:
> [Three months ago] -> [Today]
> 
> Clicking on three months ago we could choose the 'Date string' from the dropdown, which would turn the box into an editable input. When you click the input, you can either edit the date string directly, or select another fixed option from the dropdown.
> 
> Thoughts @chaoran-chen @ningxie1991 @corneliusroemer ?

@TKGZ Thanks for the suggestions. Since it seems like a lot of custom changes away from the react-modern-calendar-datepicker library has to offer. I switched to using react-datepicker library and implemented another version closer to what @corneliusroemer asks for. Can you try it out in the preview and let me know what you think?

spacephoton commented:
That's not bad, also we can edit the input.

What do you think of including the date picker only when custom range is selected, and putting the multiple choice on the left of the dates? 

Something like
[Last three months]

Then switch to
[Custom dates] [2021-09-28] -> [2021-09-29]

ningxie1991 commented:
> That's not bad, also we can edit the input.
> 
> What do you think of including the date picker only when custom range is selected, and putting the multiple choice on the left of the dates?
> 
> Something like
> [Last three months]
> 
> Then switch to
> [Custom dates] [2021-09-28] -> [2021-09-29]

I think moving the custom range to the left of the date range inputs / date picker is a good idea. But I'm not sure about displaying the date range inputs / datepicker only when user selects a custom range from the left dropdown. It might be too many steps for expert users. @corneliusroemer what do you think?

![image](https://user-images.githubusercontent.com/3387698/135062394-9de5dd3d-03f7-4780-971a-810e943690b2.png)

ningxie1991 committed:
Update DateRangeSelector.tsx

1. Moved custom range to the left of datepicker

spacephoton commented:
> I think moving the custom range to the left of the date range inputs / date picker is a good idea. But I'm not sure about displaying the date range inputs / datepicker only when user selects a custom range from the left dropdown. It might be too many steps for expert users. @corneliusroemer what do you think?

The thinking is that once you set a custom date range, you can easily change it, but it might be confusing to have many buttons on first look

corneliusroemer commented:
Perfect, we're getting there. Thanks for taking my feedback into account.

What could still be improved:
- tab navigation is broken, when I'm in `from:` and tab, I enter the date picker rather than the `to:` field, date picker should not be for key navigation, it's a point and click only tool.
- the fields are not wide enough, days are cut off (minor css tweak necessary)
![image](https://user-images.githubusercontent.com/25161793/135134864-937e6120-5ada-468b-840f-11b62950d64a.png)

spacephoton commented:
> Perfect, we're getting there. Thanks for taking my feedback into account.
> 
> What could still be improved:
> 
> * tab navigation is broken, when I'm in `from:` and tab, I enter the date picker rather than the `to:` field, date picker should not be for key navigation, it's a point and click only tool.
> * the fields are not wide enough, days are cut off (minor css tweak necessary)
>   ![image](https://user-images.githubusercontent.com/25161793/135134864-937e6120-5ada-468b-840f-11b62950d64a.png)

Seems like the cut off is from within the component so maybe just need that padding reduced 👍 

Another aesthetic thing I think would be nice is to have an arrow instead of "From" and "Until" since it conveys the same information 😄  Kind of like this:

![CleanShot 2021-09-28 at 20 01 24](https://user-images.githubusercontent.com/36269621/135141026-a67d139c-0468-4788-b0a1-847f2cdb7eec.png)

spacephoton commented:
Also really like the way the new selector works because it requires less clicks! 🥳

corneliusroemer commented:
> Also really like the way the new selector works because it requires less clicks! 🥳

That's exactly the spirit :heart: 👩‍💻

ningxie1991 committed:
Merge branch 'develop' into fine-grain-date-filter

ningxie1991 committed:
Updated date picker

ningxie1991 committed:
updated tabbing

ningxie1991 committed:
Update DateRangeSelector.tsx

ningxie1991 committed:
Update DateRangeSelector.tsx

fix build

ningxie1991 committed:
Update DateRangeSelector.tsx

fix format

corneliusroemer commented:
Can you fix the too narrow box @ningxie1991 ? This is on Brave:
![image](https://user-images.githubusercontent.com/25161793/135452744-17a9b0af-e514-41ea-9548-a85fca907bda.png)

ningxie1991 commented:
> Can you fix the too narrow box @ningxie1991 ? This is on Brave: ![image](https://user-images.githubusercontent.com/25161793/135452744-17a9b0af-e514-41ea-9548-a85fca907bda.png)

I am trying to reproduce this cutoff, but I can't see it on any screen size on my machine. How did you see this? @corneliusroemer @TKGZ 

This is what I see on Brave for a small screen size:

![image](https://user-images.githubusercontent.com/3387698/135453927-3a4a8f5e-db5e-4eef-a653-a3007a75004a.png)

On chrome:

![image](https://user-images.githubusercontent.com/3387698/135454061-cb0971a0-9b5b-4aa2-8a63-124e97278d66.png)

ningxie1991 commented:
@chaoran-chen @TKGZ Do you have any idea why Docker is failing in the checks?

spacephoton commented:
> @chaoran-chen @TKGZ Do you have any idea why Docker is failing in the checks?

We have to see with Chaoran i seem to have the same issue

chaoran-chen commented:
I like to new layout! Only on mobile, it isn't working so well:
![image](https://user-images.githubusercontent.com/18666552/135464054-ffd34707-c8d7-410f-9828-2da14bdbc498.png)

> @chaoran-chen @TKGZ Do you have any idea why Docker is failing in the checks?

I changed the Docker secret from a PAT to the new "implicit" GitHub secret in 56085e48528115fbfa1301604aa5973998a8c0e4. Merging the develop branch into this one should fix the problem.

spacephoton commented:
I also have it when I open in chrome, not sure what it could be
![CleanShot 2021-09-30 at 15 43 57](https://user-images.githubusercontent.com/36269621/135466845-0f4c0b17-d05d-4b55-9675-b07ac298ddb6.png)

spacephoton commented:
![CleanShot 2021-09-30 at 15 44 59](https://user-images.githubusercontent.com/36269621/135466995-808ab13a-d908-4b13-ba42-f36c6d1122e4.gif)

Also, custom dates is not clickable. 
![CleanShot 2021-09-30 at 15 45 52](https://user-images.githubusercontent.com/36269621/135467142-39674113-7002-431b-b982-1413316f9cb7.png)


Another issue: When on 'all times' choosing a new date doesn't do anything. I still think that we shouldn't show the dates on 'all times' just to make the UI less overwhelming.  what do you think @chaoran-chen

ningxie1991 committed:
Merge branch 'develop' into fine-grain-date-filter

chaoran-chen commented:
> Another issue: When on 'all times' choosing a new date doesn't do anything. I still think that we shouldn't show the dates on 'all times' just to make the UI less overwhelming. what do you think @chaoran-chen

I agree. Could we change the value in the dropbox menu to "Custom range" whenever the user manually changes the values in the date filter?

Further, it would be great to have Monday as the first day again (to be consistent with the ISO week).

ningxie1991 committed:
Update DateRangeSelector.tsx

fix bug

ningxie1991 commented:
> ![CleanShot 2021-09-30 at 15 44 59](https://user-images.githubusercontent.com/36269621/135466995-808ab13a-d908-4b13-ba42-f36c6d1122e4.gif)
> 
> Also, custom dates is not clickable. ![CleanShot 2021-09-30 at 15 45 52](https://user-images.githubusercontent.com/36269621/135467142-39674113-7002-431b-b982-1413316f9cb7.png)
> 
> Another issue: When on 'all times' choosing a new date doesn't do anything. I still think that we shouldn't show the dates on 'all times' just to make the UI less overwhelming. what do you think @chaoran-chen

Ohh this is a bug from trying to resolve the broken tab action that @corneliusroemer mentioned. I rolled it back now. But I still can't figure out how to get tabbing to work the way @corneliusroemer wanted. 

Now it's like this even after setting `enableTabLoop={false}` :  

![A661EDB5-F016-4998-9969-D46E4944D7C6](https://user-images.githubusercontent.com/3387698/135470346-96063bc2-5c4e-44c8-b9be-a80812bc9369.GIF)

chaoran-chen commented:
If tabbing does not want to work, we can also deal with it later and create an issue.

ningxie1991 committed:
Added some more attributes to datepicker

ningxie1991 commented on src/components/DateRangeSelector.tsx:
```diff
@@ -0,0 +1,136 @@
+              enableTabLoop={false}
+              disabledKeyboardNavigation={true}
+              preventOpenOnFocus={true}
+              className='border rounded py-1.5 px-3 w-32 focus:outline-none focus:ring focus:border-blue-200'
```
@TKGZ This is the input style, maybe you can play around with it to see how the input won't get cutoff on your screen?

corneliusroemer replied:
You should just get rid of all right padding.

The problem seems to be: the width is fixed to xREM, when it should be adjustd to length of text.

Plus: you have right padding which in case of too narrow box will cut off text from the right.

So set right padding to 0. And then possibly make the box width better adjusted. Min x but possibly longer if necessary.

ningxie1991 replied:
But how come it appears normal on my screen than yours? I am just puzzled by this difference....

ningxie1991 commented:
> If tabbing does not want to work, we can also deal with it later and create an issue.

Alright, I increased the width of the input box hopefully it will solve the cutoff issue that @TKGZ and @corneliusroemer sees. Even though now the input boxes look like this on my screen as they are pretty long:

![image](https://user-images.githubusercontent.com/3387698/135474419-5f5ecafb-f7ea-47c7-9945-2093df289b54.png)

ningxie1991 committed:
removed unused import and right padding

chaoran-chen commented on src/components/DateFilter.tsx:
```diff
@@ -0,0 +1,218 @@
+    ],
+
+    // just play around with this number between 0 and 6
+    weekStartingIndex: 0,
```
Can we set this to Monday, please? :)

ningxie1991 replied:
Actually this file is for the first version of the date filter, the one that we discarded. But I will check how to do that for the new data range selector (the one with from and to input boxes). I should delete this file.

ningxie1991 replied:
> Can we set this to Monday, please? :)

It's done.

ningxie1991 committed:
Merge branch 'develop' into fine-grain-date-filter

ningxie1991 committed:
Deleted DateFilter.tsx file and changed calendar to start on Monday

ningxie1991 committed:
Fix formatting issue

ningxie1991 committed:
Update package-lock.json

Updated npm to latest version 8.0.0

ningxie1991 commented:
I updated my npm to latest version 8.0.0. @chaoran-chen @TKGZ
